src/reducers/index.js

```diff
@@ -0,0 +1,64 @@
+const initialState = {
+  posts: [],
+  post: null,
+  isLoaded: false,
+  error: false,
+  isLoggedOn: false,
+  loginButtonValue: 'Login'
+};
+
+const reducer = (state = initialState, action) => {
+  switch (action.type) {
+    case 'FETCH_POSTS_REQUEST':
+      return {
+        ...state
+      };
+
+    case 'FETCH_POSTS_SUCCESS':
+      return {
+        ...state,
+        posts: action.payload,
+        isLoaded: true,
+        error: false
+      };
+    case 'FETCH_POSTS_ERROR':
+      return {
+        ...state,
+        posts: [],
+        isLoaded: false,
+        error: true
+      };
+
+    case 'FETCH_POST_DETAILS_REQUEST':
+      return {
+        ...state
+      };
+
+    case 'FETCH_POST_DETAILS_SUCCESS':
+      return {
+        ...state,
+        isLoaded: true,
+        post: action.payload
+      };
+
+    case 'FETCH_POST_DETAILS_FAILURE':
+      return {
+        ...state,
+        post: null,
+        error: true
+      };
+
+    case 'PERFORM_LOGIN':
+      return {
+        ...state,
+        isLoggedOn: (state.isLoggedOn = !state.isLoggedOn),
```

Comment:
Mutation of state? WTF? :)))

```suggestion
        isLoggedOn: !state.isLoggedOn,
```